lab_5_scrapper/scrapper.py

```diff
@@ -13,70 +35,112 @@ def __init__(self, path_to_config: Path) -> None:
+            if not isinstance(url, str):
+                raise IncorrectSeedURLError('seed URL is not str')
+            if not re.match(regex, url):
+                raise IncorrectSeedURLError('seed URL does not match standard pattern "https?://w?w?w?."')
+        if not isinstance(self.config_dict['total_articles_to_find_and_parse'], int) or \
+                self.config_dict['total_articles_to_find_and_parse'] <= 0:
+            raise IncorrectNumberOfArticlesError('total number of articles to parse is not integer')
+        if not 1 <= self.config_dict['total_articles_to_find_and_parse'] <= 150:
+            raise NumberOfArticlesOutOfRangeError('total number of articles is out of range from 1 to 150')
+        if not isinstance(self.config_dict['headers'], dict):
+            raise IncorrectHeadersError('headers are not in a form of dictionary')
+        if not isinstance(self.config_dict['encoding'], str):
+            raise IncorrectEncodingError('encoding must be specified as a string')
+        if not isinstance(self.config_dict['timeout'], int) or not 0 <= self.config_dict['timeout'] <= 60:
+            raise IncorrectTimeoutError('timeout value must be a positive integer less than 60')
+        if not isinstance(self.config_dict['should_verify_certificate'], bool):
+            raise IncorrectVerifyError('verify certificate value must either be True or False')
+        if not isinstance(self.config_dict['headless_mode'], bool):
+            raise IncorrectVerifyError('headless mode value must either be True or False')
 
     def get_seed_urls(self) -> list[str]:
         """
         Retrieve seed urls
         """
-        pass
+        return self._seed_urls
 
     def get_num_articles(self) -> int:
         """
         Retrieve total number of articles to scrape
         """
-        pass
+        return self._num_articles
 
     def get_headers(self) -> dict[str, str]:
         """
         Retrieve headers to use during requesting
         """
-        pass
+        return self._headers
 
     def get_encoding(self) -> str:
         """
         Retrieve encoding to use during parsing
         """
-        pass
+        return self._encoding
 
     def get_timeout(self) -> int:
         """
         Retrieve number of seconds to wait for response
         """
-        pass
+        return self._timeout
 
     def get_verify_certificate(self) -> bool:
         """
         Retrieve whether to verify certificate
         """
-        pass
+        return self._should_verify_certificate
 
     def get_headless_mode(self) -> bool:
         """
         Retrieve whether to use headless mode
         """
-        pass
+        return self._headless_mode
 
 
 def make_request(url: str, config: Config) -> requests.models.Response:
     """
     Delivers a response from a request
     with given configuration
     """
-    pass
+    return requests.get(url, headers=config.get_headers())
```

Comment:
you also need to set encoding to response

Reply:
Done